ocean_lib/common/ddo/ddo.py

```diff
@@ -354,3 +387,64 @@ def create_authentication_from_json(values):
 
         authentication = {"type": authentication_type, "publicKey": key_id}
         return authentication
+
+    def get_addresses_of_type(self, section_type="allow"):
+        """Get a filtered list of addresses from credentials (use with allow/deny)."""
+        entries = self._credentials.get(section_type, [])
+
+        for entry in entries:
+            if entry["type"] == "address":
+                return [val.lower() for val in entry["values"]]
+
+        return []
+
+    def enable(self):
+        if not self._status:
+            self._status = {}
+
+        self._status.pop("isOrderDisabled")
+
+    def disable(self):
+        if not self._status:
+            self._status = {}
+
+        self._status["isOrderDisabled"] = True
+
+    def get_address_allowed_code(self, credential=None):
```

Comment:
I think it makes it much more easier to read, by separating the concerns. One function handles high-level access granting, while this one is a helper. No, it should not be called outside is_consumable, but we don't have protected/private in Python :(

Reply:
Fair enough. In that case, I recommend prepending an underscore like so:  `_get_allowed_code` to hint that this method is meant for internal use. I based this name on the latest changes which have already removed the "address" part of the method name

Reply:
I'm resolving this comment because this method is no longer internal because it's now located in `credentials.py` but it's called from `ddo.py`. Thus the prepended underscore no longer makes sense.